rewrite-java/src/main/java/org/openrewrite/java/ChangeType.java

```diff
@@ -97,6 +105,16 @@ public TreeVisitor<?, ExecutionContext> getVisitor() {
                     if (!Boolean.TRUE.equals(ignoreDefinition) && containsClassDefinition(cu, oldFullyQualifiedTypeName)) {
                         return SearchResult.found(cu);
                     }
+                    if (Boolean.TRUE.equals(visitStringLiterals)) {
+                        return Traits.literal().asVisitor(lit -> {
```

Comment:
I've pushed a new test that fails: when `visitStringLiterals = true`, but there is no String literal match, then we now also fail to replace type matches. I'd have expected this recipe to continue making type changes and not _only_ String literals. The same applies for ChangePackage, but I've not added a test there.